src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NServiceBus/LoadHandlersConnectorWrapper.cs

```diff
@@ -61,6 +59,10 @@ void OnComplete(Task task)
                 {
                     transaction.End();
                 }
+                else if (task?.Status == TaskStatus.Faulted)
+                {
+                    transaction.NoticeError(task.Exception);
```

Comment:
Do we still want to end the transaction after noticing the error, or does that happen somewhere else? 

I reviewed some of the other handlers and it looks like the default delegates, in general, set segment.ErrorData instead of calling NoticeError. I'm wondering if calling NoticeError is correct?

Reply:
Nice find and good question!  I don't think it will end on its own.  I copied how the ReceiveMessageWrapper captured the error.  Let me take a look and see what other wrappers are doing.

Reply:
Okay. 

ending the transaction: looking at the GetDelegateFor method, it will call all three actions (onSuccess, onFailure, onComplete) in that order.  onComplete is always called if the Action is not null. Since we handle ending the the onComplete for this wrapper, we are good.

NoticeError vs ErrorData:  I checked out a few other wrappers with RabbitMQ being the closest and it uses the same pattern - NoticeError.  I saw a few others that did things a bit different, but this seems to be a pattern we are using in places.

Thoughts?

Reply:
Thanks for the information. This satisfies my curiosity.